chart/templates/crds/chaos.datadoghq.com_disruptions.yaml

```diff
@@ -407,6 +407,26 @@ spec:
                           type: string
                         namespace:
                           type: string
+                        ports:
+                          items:
+                            properties:
+                              name:
+                                type: string
+                              port:
+                                maximum: 65535
+                                minimum: 0
+                                type: integer
+                              protocol:
+                                enum:
+                                - tcp
+                                - udp
+                                - sctp
+                                - ""
```

Comment:
Could we set a default instead?
To make it explicit to our users we will use tcp when value is not specified?

Reply:
To me the default value means we don't try to find a servicePort with an associated protocol, so it can be anything.
We actually don't use the value except for filtering / finding the servicePort, so it makes no sense to me having a default. Meaning that if I specify this:

```
port: 8080
name: hello-world
```

and I have this set up in my service:
```
port: 8080
name: hello-world
protocol: udp
```

it finds this servicePort.

WDYT?

Reply:
👍 ok 😅 

So, let me try to rephrase to ensure I understood:
* If I do not specify a protocol value, the service port associated, no matter it's protocol, will be "matched"
* If I specify a protocol value, the service port associated need to have specified protocol to be "matched"

🤔 

With this new enlightening I'm wondering why we should have protocol in the end (I know, it's me that suggested it...)

It seems it provide a filtering layer that is useless IMO (protect against an unexpected port protocol change?)

It feels, it's overly complex in the end, for a relative added value nobody asked for.

Hence, I would nearly get back to your original implementation, a simple array...

Except, this array could contain `intOrString` int being the port value, string being the port name.

And the only task we do, is we match against this array values (either port or name)

The protocol is automatically retrieved from the Kubernetes service definition (and we might emphasis this, as I forgot it during the review TBH).

Sorry for the back and forth, and THANKS for highlighting the misunderstanding I had 🤗

Reply:
If you have a strong opinion on keeping the protocol to provide additional filtering capabilities, I'm also fine with it 🤗 

Just wanted to highlight my comprehension and challenge the benefits from my POV

Reply:
I agree with you on the fact that maybe specifying the protocol is overkill 😅. I would however prefer to have a servicePort definition with an optional name and/or port instead of having an array of either port or names. I feel like it would be clearer for the user but maybe having the added logic of specifying name: or port: would be even more confusing? What do you think?

So, it would look like:

```
services:
  - name: demo # service name
    namespace: chaos-demo # service namespace
    servicePorts:
       - port: 8080
       - name: special-port
```

instead of:

```
services:
  - name: demo # service name
    namespace: chaos-demo # service namespace
    servicePorts: 
       - 8080
       - special-port
```

I'll also add more documentation.

Reply:
Is it possible in k8s to have a service set up that re-uses the same port but for different protocols?

Reply:
> Is it possible in k8s to have a service set up that re-uses the same port but for different protocols?

At least nothing is preventing you to do so (tested locally)

That imply the destination application is able to differentiate traffic received on the same port (multiplexer?), I'm not sure it's somehow possible 🤔 I do not know TBH.

The suggested implementation (without protocol but with a dedicated object) enables us to extend the implementation later if we have feedbacks requiring such use case.

Would it be enough for you?